src/openrct2/actions/BannerSetStyleAction.hpp

```diff
@@ -0,0 +1,185 @@
+                if (_parameter > 13)
+                {
+                    log_error("Invalid text colour: colour = %u", _parameter);
+                    return MakeResult(GA_ERROR::INVALID_PARAMETERS, STR_CANT_REPAINT_THIS);
+                }
+                break;
+            case BannerSetStyleType::NoEntry:
+                break;
+            default:
+                log_error("Invalid type: %u", _type);
+                return MakeResult(GA_ERROR::INVALID_PARAMETERS, STR_NONE);
+        }
+        return res;
+    }
+
+    GameActionResult::Ptr Execute() const override
+    {
+        auto res = MakeResult();
+
+        rct_banner* banner = &gBanners[_bannerIndex];
+
+        res->ExpenditureType = RCT_EXPENDITURE_TYPE_LANDSCAPING;
+        res->Position.x = banner->x * 32 + 16;
+        res->Position.y = banner->y * 32 + 16;
+        res->Position.z = tile_element_height(banner->x, banner->y) & 0xFFFF;
+
+        TileElement* tileElement = banner_get_tile_element(_bannerIndex);
+
+        if (tileElement == nullptr)
+        {
+            log_error("Could not find banner index = %u", _bannerIndex);
+            return MakeResult(GA_ERROR::INVALID_PARAMETERS, STR_NONE);
+        }
+
+        switch (static_cast<BannerSetStyleType>(_type))
+        {
+            case BannerSetStyleType::PrimaryColour:
+                banner->colour = _parameter;
+                break;
+            case BannerSetStyleType::TextColour:
+            {
+                banner->text_colour = _parameter;
+                int32_t colourCodepoint = FORMAT_COLOUR_CODE_START + banner->text_colour;
+
+                utf8 buffer[256];
+                format_string(buffer, 256, banner->string_idx, nullptr);
+                int32_t firstCodepoint = utf8_get_next(buffer, nullptr);
+                if (firstCodepoint >= FORMAT_COLOUR_CODE_START && firstCodepoint <= FORMAT_COLOUR_CODE_END)
+                {
+                    utf8_write_codepoint(buffer, colourCodepoint);
+                }
+                else
+                {
+                    utf8_insert_codepoint(buffer, colourCodepoint);
+                }
+
+                rct_string_id stringId = user_string_allocate(USER_STRING_DUPLICATION_PERMITTED, buffer);
+                if (stringId != 0)
+                {
+                    rct_string_id prevStringId = banner->string_idx;
+                    banner->string_idx = stringId;
+                    user_string_free(prevStringId);
+                }
+                else
+                {
+                    return MakeResult(GA_ERROR::INVALID_PARAMETERS, STR_ERR_CANT_SET_BANNER_TEXT);
+                }
+                break;
+            }
+            case BannerSetStyleType::NoEntry:
+            {
+                BannerElement* bannerElement = tileElement->AsBanner();
+                banner->flags &= BANNER_FLAG_NO_ENTRY;
+                banner->flags |= BANNER_FLAG_NO_ENTRY & (_parameter != 0);
```

Comment:
```suggestion
                banner->flags |= (parameter != 0) ? BANNER_FLAG_NO_ENTRY : 0
```
Would be slightly easier to read.